job_executor/app.py

```diff
@@ -268,9 +294,12 @@ def clean_up_after_dead_workers(dead_workers: List[Worker]) -> None:
             if job and job.status not in ["queued", "built"]:
                 logger.info(f"Worker died and did not finish job {job.job_id}")
```

Comment:
Not in the scope of this PR, but maybe we should open a refactor/housekeeping task: This should probably be an ERROR level log